### PR TITLE
allow routing to/from names

### LINKED ISSUE #22: Combine routing api and geocoder to allow routing to/from names
Add a new function to the routing api that takes names for the origin and destination.
Call the geocoder api to resolve these to coordinates.

I would be happy to create a PR for this if you would welcome this feature.

### PR DESCRIPTION
**Pull request by eifinger**
Closes #22

### CODE REVIEW

**Comment by eifinger** [comment]
@abdullahselek Codecov upload fails

**Comment by abdullahselek** [comment]
> @abdullahselek Codecov upload fails

I realized I did something wrong when I setup codecov on workflows. Could you please rebase from `master` into your branch?

File: herepy/routing_api.py [hunk 1]
```diff
@@ -84,16 +89,18 @@ def _route(self, waypoint_a, waypoint_b, modes=None, departure=None, arrival=Non
...
-                      waypoint_b: List[float],
+                      waypoint_a: Union[List[float], str],
+                      waypoint_b: Union[List[float], str],
                       modes: List[RouteMode]=None,
                       departure: str='now'):
         """Request a bicycle route between two points
         Args:
-          waypoint_a (array):
+          waypoint_a:
             array including latitude and longitude in order.
```
Comment (abdullahselek) [inline herepy/routing_api.py]: Please remove the `.` at the end.

File: herepy/routing_api.py [hunk 2]
```diff
@@ -84,16 +89,18 @@ def _route(self, waypoint_a, waypoint_b, modes=None, departure=None, arrival=Non
...
-          waypoint_b (array):
+            or string with the location name.
+          waypoint_b:
             array including latitude and longitude in order.
```
Comment (abdullahselek) [inline herepy/routing_api.py]: Same here, please remove the `.` at the end.

File: herepy/routing_api.py [hunk 3]
```diff
@@ -108,16 +115,18 @@ def bicycle_route(self,
...
-                  waypoint_a: List[float],
-                  waypoint_b: List[float],
+                  waypoint_a: Union[List[float], str],
+                  waypoint_b: Union[List[float], str],
                   modes: List[RouteMode]=None,
                   departure: str='now'):
         """Request a driving route between two points
         Args:
           waypoint_a (array):
             array including latitude and longitude in order.
```
Comment (abdullahselek) [inline herepy/routing_api.py]: Same in other docs which parameter type changes as `Union`.

File: herepy/routing_api.py [hunk 4]
```diff
@@ -108,16 +115,18 @@ def bicycle_route(self,
...
+            or string with the location name.
           waypoint_b (array):
             array including latitude and longitude in order.
```
Comment (abdullahselek) [inline herepy/routing_api.py]: same here.

File: herepy/routing_api.py [hunk 5]
```diff
@@ -132,16 +141,18 @@ def car_route(self,
...
-                         waypoint_a: List[float],
-                         waypoint_b: List[float],
+                         waypoint_a: Union[List[float], str],
+                         waypoint_b: Union[List[float], str],
                          modes: List[RouteMode]=None,
                          departure: str='now'):
         """Request a pedastrian route between two points
         Args:
           waypoint_a (array):
             array including latitude and longitude in order.
```
Comment (abdullahselek) [inline herepy/routing_api.py]: Not to forget :)

File: herepy/routing_api.py [hunk 6]
```diff
@@ -132,16 +141,18 @@ def car_route(self,
...
+            or string with the location name.
           waypoint_b (array):
             array including latitude and longitude in order.
```
Comment (abdullahselek) [inline herepy/routing_api.py]: Not to forget :)

File: herepy/routing_api.py [hunk 7]
```diff
@@ -156,19 +167,22 @@ def pedastrian_route(self,
...
-                           waypoint_c: List[float],
+                           waypoint_a: Union[List[float], str],
+                           waypoint_b: Union[List[float], str],
+                           waypoint_c: Union[List[float], str],
                            modes: List[RouteMode]=None,
                            departure: str='now'):
         """Request a intermediate route from three points
         Args:
           waypoint_a (array):
             Starting array including latitude and longitude in order.
```
Comment (abdullahselek) [inline herepy/routing_api.py]: For every parameter which is now a `Union`.

**Comment by eifinger** [comment]
Should I remove both `.` or only the ones after "order"?

**Comment by abdullahselek** [comment]
> Should I remove both `.` or only the ones after "order"?

Only the ones after "order" please.

**Comment by eifinger** [comment]
@abdullahselek great thanks. Can you release a new version?

**Comment by abdullahselek** [comment]
Yes I am going to release new version.

**Comment by abdullahselek** [comment]
Version `2.1.0` is now available, thanks for your contribution.

**Pull request merged**